src/ducks/sharing/index.js

```diff
@@ -1,12 +1,15 @@
 /* global cozy */
-export const filterSharedByMeDocuments = (ids, doctype) =>
-  findPermissionSets(ids, doctype, SHARED_BY_LINK).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+export const filterSharedByLinkDocuments = (ids, doctype) =>
+  findPermissionSets(ids, doctype, SHARED_BY_LINK, 'collection').then(sets => sets.map(set => set.attributes.permissions['collection'].values[0]))
 
 export const filterSharedWithMeDocuments = (ids, doctype) =>
-  findPermissionSets(ids, doctype, SHARED_WITH_ME).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+  findPermissionSets(ids, doctype, SHARED_WITH_ME, 'collection').then(sets => sets.map(set => set.attributes.permissions['collection'].values[0]))
 
-export const findPermSet = (id, doctype) =>
-  findPermissionSets([id], doctype, SHARED_BY_LINK).then(sets => sets.length === 0 ? undefined : sets[0])
+export const filterSharedWithOthersDocuments = (id, doctype) =>
+  findPermissionSets([id], doctype, SHARED_WITH_OTHERS, 'rule0')
```

Comment:
Are you sure it's safe to use `rule0` here? IIRC there can be multiple rules in a permission set, and there's no guarantee that the ule we're looking for is the first one?